### PR TITLE
perf: Set stdout and stderr streams of perf process to non blocking

### PR DESCRIPTION
**Pull request by mpozniak95**
## Description
This PR sets stdout and stderr streams of perf process to non blocking.

## Related Issue
https://github.com/Granulate/gprofiler/issues/612

## How Has This Been Tested?
Added `raise exception()` to try block in `wait_and_script()` -> on master - gProfiler stucks on reading from stdout and stderr (like in related issue), after these changes it reads the stream and continues

### CODE REVIEW

File: gprofiler/profilers/perf.py [hunk 1]
```diff
@@ -107,6 +107,8 @@ def wait_and_script(self) -> str:
             perf_data = wait_for_file_by_prefix(f"{self._output_path}.", self._dump_timeout_s, self._stop_event)
         except Exception:
             assert self._process is not None and self._process.stdout is not None and self._process.stderr is not None
+            os.set_blocking(self._process.stdout.fileno(), False)
```
Comment (Jongy) [inline gprofiler/profilers/perf.py:110]: Let's move these 2 to the `start()` method, not here. Can do that in the `else` branch in `start()`.

File: gprofiler/profilers/perf.py [hunk 2]
```diff
@@ -107,6 +107,8 @@ def wait_and_script(self) -> str:
...
+            os.set_blocking(self._process.stderr.fileno(), False)
             logger.critical(
                 f"perf failed to dump output. stdout {self._process.stdout.read()!r}"
```
Comment (Jongy) [inline gprofiler/profilers/perf.py:113]: What happens to this `read()` when it's nonblocking? We get back an empty string?

Comment (mpozniak95) [reply gprofiler/profilers/perf.py:113]: We get None

Comment (Jongy) [reply gprofiler/profilers/perf.py]: kk

**Comment by Jongy** [comment]
I managed to reproduce with this diff:
```
diff --git a/gprofiler/profilers/perf.py b/gprofiler/profilers/perf.py
index 990e077e..a385d8b1 100644
--- a/gprofiler/profilers/perf.py
+++ b/gprofiler/profilers/perf.py
@@ -264,8 +264,8 @@ class SystemProfiler(ProfilerBase):
         if self._stop_event.wait(self._duration):
             raise StopEventSetException
 
-        for perf in self._perfs:
-            perf.switch_output()
+        # for perf in self._perfs:
+        #     perf.switch_output()
 
         return {
             k: self._generate_profile_data(v, k)
```

we don't send a signal, so perf doesn't write the output file, then we time-out in `wait_for_file_by_prefix` and eventually I get the same hang in
```
Thread 0x7F11117FA700 (active): "ThreadPoolExecutor-0_4"
    wait_and_script (gprofiler/profilers/perf.py:111)
    snapshot (gprofiler/profilers/perf.py:273)
    run (concurrent/futures/thread.py:57)
    _worker (concurrent/futures/thread.py:80)
    run (threading.py:870)
    _bootstrap_inner (threading.py:932)
    _bootstrap (threading.py:890)
```

On your branch - the same diff doesn't hang and just keeps erroring. Assuming the problem is interleaved, it will repair itself I bet.

**Review by Jongy — APPROVED** [review]

**Pull request merged**